### PR TITLE
React: Support deleting inbox notifications

### PR DESCRIPTION
**Pull request by marcbouchenoire**
This PR adds support for deleting inbox notifications in `@liveblocks/react` and `@liveblocks/react-ui`:
- Add `useDeleteInboxNotification` and `useDeleteAllInboxNotifications` hooks
- Add "Delete notification" action to `InboxNotification` default component
- Update documentation

Fixes LB-981
Fixes LB-983

### CODE REVIEW

File: docs/pages/api-reference/liveblocks-react.mdx [hunk 1]
```diff
@@ -2477,6 +2477,28 @@ const markAllInboxNotificationsAsRead = useMarkAllInboxNotificationsAsRead();
 markAllInboxNotificationsAsRead();
 ```
 
+### useDeleteInboxNotification [@badge=LiveblocksProvider]
+
+Returns a function that deletes an inbox notification.
```
Comment (CTNicholas) [inline docs/pages/api-reference/liveblocks-react.mdx:2482]:
```suggestion
Returns a function that deletes an inbox notification for the current user.
```

This only works for the current user's notifications, right? Maybe should clarify that

File: docs/pages/api-reference/liveblocks-react.mdx [hunk 2]
```diff
@@ -2477,6 +2477,28 @@ const markAllInboxNotificationsAsRead = useMarkAllInboxNotificationsAsRead();
...
+```tsx
+import { useDeleteInboxNotification } from "@liveblocks/react/suspense";
+
+const deleteInboxNotification = useDeleteInboxNotification();
+deleteInboxNotification("in_xxx");
+```
+
+### useDeleteAllInboxNotifications [@badge=LiveblocksProvider]
+
+Returns a function that deletes all inbox notifications.
```
Comment (CTNicholas) [inline docs/pages/api-reference/liveblocks-react.mdx:2493]:
```suggestion
Returns a function that deletes all of the current user‘s inbox notifications.
```

File: docs/pages/api-reference/liveblocks-react.mdx (hunk 1 above)
Comment (marcbouchenoire) [reply docs/pages/api-reference/liveblocks-react.mdx:2482]: Yep technically inbox notifications are user-based so you will always only see yours but I'll clarify 👍

File: packages/liveblocks-react/test-d/jsdocs.test-d.ts [hunk 3]
```diff
@@ -167,6 +171,8 @@ expectDocCommentIncludes<_>(suspense.ClientSideSuspense({} as any));
 // suspense.useLostConnectionListener
 // suspense.useMarkAllInboxNotificationsAsRead
 // suspense.useMarkInboxNotificationAsRead
+// suspense.useDeleteAllInboxNotifications
+// suspense.useDeleteInboxNotification
```
[the remark this answers is not in the document]
Comment (marcbouchenoire) [reply packages/liveblocks-react/test-d/jsdocs.test-d.ts:175]: Nice, I wasn't sure, will do! 🧹

**Pull request merged**